cpp/src/groupby/new_groupby.cu

```diff
@@ -190,3 +192,185 @@ gdf_error gdf_group_by(gdf_column* in_key_columns[],
 
   return gdf_error_code;
 }
+
+
+
+gdf_error gdf_unique_indices(gdf_size_type num_data_cols,
+                     gdf_column** data_cols_in,
+									   gdf_index_type* unique_indices,
+                     gdf_size_type* num_unique_indices, 
+									   gdf_context* ctxt)
+{
+
+  gdf_size_type nrows = data_cols_in[0]->size;
+  // setup for reduce by key  
+  bool have_nulls = false;
+  for (int i = 0; i < num_data_cols; i++) {
+    if (data_cols_in[i]->null_count > 0) {
+      have_nulls = true;
+      break;
+    }
+  }
+
+  rmm::device_vector<void*> d_cols(num_data_cols); 
+  rmm::device_vector<int> d_types(num_data_cols, 0);
+  void** d_col_data = d_cols.data().get();
+  int* d_col_types = d_types.data().get();
+
+  bool nulls_are_smallest = ctxt->flag_nulls_sort_behavior == GDF_NULL_AS_SMALLEST;
+
+  gdf_index_type* result_end;
+  cudaStream_t stream;
+  cudaStreamCreate(&stream);
+  auto exec = rmm::exec_policy(stream)->on(stream);
+  if (have_nulls){
+
+    rmm::device_vector<gdf_valid_type*> d_valids(num_data_cols);
+    gdf_valid_type** d_valids_data = d_valids.data().get();
+
+    soa_col_info(data_cols_in, num_data_cols, d_col_data, d_valids_data, d_col_types);
+    
+    LesserRTTI<gdf_size_type> comp(d_col_data, d_valids_data, d_col_types, nullptr, num_data_cols, nulls_are_smallest);
+    
+    auto counting_iter = thrust::make_counting_iterator<gdf_size_type>(0);
+    
+    result_end = thrust::unique_copy(exec, counting_iter, counting_iter+nrows, 
+                              unique_indices,
+                              [comp]  __device__(gdf_size_type key1, gdf_size_type key2){
+                              return comp.equal_with_nulls(key1, key2);
+                            });
+
+  } else {
+
+    soa_col_info(data_cols_in, num_data_cols, d_col_data, nullptr, d_col_types);
+    
+    LesserRTTI<gdf_size_type> comp(d_col_data, nullptr, d_col_types, nullptr, num_data_cols, nulls_are_smallest);
+
+    auto counting_iter = thrust::make_counting_iterator<gdf_size_type>(0);
+    
+    result_end = thrust::unique_copy(exec, counting_iter, counting_iter+nrows, 
+                              unique_indices,
+                              [comp]  __device__(gdf_size_type key1, gdf_size_type key2){
+                              return comp.equal(key1, key2);  
+                            });
+  }
+
+  gdf_size_type new_sz = thrust::distance(unique_indices, result_end);
```

Comment:
```suggestion
  gdf_size_type new_sz = thrust::distance(exec, unique_indices, result_end);
```

Reply:
thrust::distance does not take exec as a parameter

Reply:
Ah, yeah, that was just a knee-jerk reaction. Please ignore :)